src/Waives.Pipelines/DocumentProcessor.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Waives.Pipelines
+{
+    internal class DocumentProcessor : IDocumentProcessor
+    {
+        private readonly Func<Document, Task<WaivesDocument>> _docCreator;
+        private readonly IEnumerable<Func<WaivesDocument, Task<WaivesDocument>>> _docActions;
+        private readonly Action<Exception, Document> _onDocumentException;
+
+        public DocumentProcessor(Func<Document, Task<WaivesDocument>> docCreator,
```

Comment:
✋ There's an `IHttpDocumentFactory` interface that provides essentially this interface. We should either update that interface to return a `Task<WaivesDocument>` rather than a `Task<IHttpDocument>` (although I think there was a reason -- probably testability -- for separating the two ideas), or pass the factory here and do the conversion to a `WaivesDocument` where the factory is called.